### PR TITLE
Fix/rich previews

### PR DESCRIPTION
**Pull request by vvandijck**
The 'rich previews' were no longer working because of some CSP and dependency issues. Spend time trying to fix the CSP issues but came to the conclusion that I was running in circles.

It seems when they switched over to V3, they tightened their security policies even more. Loading external scripts from an `<iframe>` doesn't play well with the way Google Extensions want you to handle content security and resource integrity.

The best option was to convert and replace the `<iframe>` previews with normal Vue components.

I also put back all the Google previews that were previously removed by (https://github.com/voorhoede/heads-up/pull/272/). Some of our DatoCMS plugins use these previews. That was broken, and should now be restored once this has been merged.

## What change(s) were made
- Fixed dependency errors.
- Replaced `<iframe>` previews with a Vue component replacement. (iframe caused CSP issue)
- Completely remove the `rich-previews` directory since it's no longer needed.
- Fixed an issue where the background(color) would not be as high as the panel itself.

## Checks
- [x] Checked browser extension (light & dark theme).
- [x] Checked web app.

### CODE REVIEW

**Comment by Siilwyn** [comment]
> Some of our DatoCMS plugins use these previews. That was broken, and should now be restored once this has been merged.
Which plugins specifically? And in what way did they break? Since we concluded earlier it's best to remove the google preview I'm wary of adding it back.

File: projects/@shared/components/rich-previews/google-article-desktop-preview.vue [hunk 1]
```diff
@@ -0,0 +1,315 @@
+<template>
+  <div class="google-article-desktop-preview">
+    <h1 class="iframe-heading" :class="{ 'iframe-heading--dark-mode': isDarkMode }">NewsArticle</h1>
```
Comment (Jamerrone) [inline projects/@shared/components/rich-previews/google-article-desktop-preview.vue]: We probably don't want to use a `h1` tag any more, it was fine inside an iFrame but not in a component.

File: projects/@shared/components/rich-previews/google-article-mobile-preview.vue [hunk 2]
```diff
@@ -0,0 +1,273 @@
+<template>
+  <div class="google-article-mobile-preview">
+    <h1 class="iframe-heading" :class="{ 'iframe-heading--dark-mode': isDarkMode }">NewsArticle</h1>
```
Comment (Jamerrone) [inline projects/@shared/components/rich-previews/google-article-mobile-preview.vue]: We probably don't want to use a `h1` tag any more, it was fine inside an iFrame but not in a component.

File: projects/@shared/components/rich-previews/google-breadcrumb-preview.vue [hunk 3]
```diff
@@ -0,0 +1,214 @@
+<template>
+  <div class="google-breadcrumb-preview">
+    <h1 class="iframe-heading" :class="{ 'iframe-heading--dark-mode': isDarkMode }">Breadcrumb</h1>
```
Comment (Jamerrone) [inline projects/@shared/components/rich-previews/google-breadcrumb-preview.vue]: We probably don't want to use a `h1` tag any more, it was fine inside an iFrame but not in a component.

File: projects/@shared/components/rich-previews/google-course-preview.vue [hunk 4]
```diff
@@ -0,0 +1,321 @@
+<template>
+  <div class="google-course-preview">
+    <h1 class="iframe-heading" :class="{ 'iframe-heading--dark-mode': isDarkMode }">Course</h1>
```
Comment (Jamerrone) [inline projects/@shared/components/rich-previews/google-course-preview.vue]: We probably don't want to use a `h1` tag any more, it was fine inside an iFrame but not in a component.

File: projects/@shared/components/rich-previews/google-product-desktop-preview.vue [hunk 5]
```diff
@@ -0,0 +1,403 @@
+<template>
+  <div class="google-product-desktop-preview">
+    <h1 class="iframe-heading" :class="{ 'iframe-heading--dark-mode': isDarkMode }">Product</h1>
```
Comment (Jamerrone) [inline projects/@shared/components/rich-previews/google-product-desktop-preview.vue]: We probably don't want to use a `h1` tag any more, it was fine inside an iFrame but not in a component.

File: projects/@shared/components/rich-previews/google-product-mobile-preview.vue [hunk 6]
```diff
@@ -0,0 +1,370 @@
+<template>
+  <div class="google-product-mobile-preview">
+    <h1 class="iframe-heading" :class="{ 'iframe-heading--dark-mode': isDarkMode }">Product</h1>
```
Comment (Jamerrone) [inline projects/@shared/components/rich-previews/google-product-mobile-preview.vue]: We probably don't want to use a `h1` tag any more, it was fine inside an iFrame but not in a component.

File: projects/@shared/components/rich-previews/google-review-preview.vue [hunk 7]
```diff
@@ -0,0 +1,349 @@
+<template>
+  <div class="google-review-preview">
+    <h1 class="iframe-heading" :class="{ 'iframe-heading--dark-mode': isDarkMode }">Critic review</h1>
```
Comment (Jamerrone) [inline projects/@shared/components/rich-previews/google-review-preview.vue]: We probably don't want to use a `h1` tag any more, it was fine inside an iFrame but not in a component.

**Review by Jamerrone — CHANGES_REQUESTED** [review]
Looking good, I pretty much only have 1 feedback. A lot of the Google components have a `h1` tag in them, that was fine as an iFrame but not so much as a component. We don't want to render multiple `h1`'s in a single page.

**Review by Jamerrone — APPROVED** [review]

**Pull request merged**